src/sentry/tasks/backfill_seer_grouping_records.py

```diff
@@ -0,0 +1,338 @@
+from datetime import datetime, timedelta
+from typing import Any, TypedDict
+
+import sentry_sdk
+from django.conf import settings
+from google.api_core.exceptions import DeadlineExceeded, ServiceUnavailable
+from snuba_sdk import Column, Condition, Entity, Function, Op, Query, Request
+from snuba_sdk.orderby import Direction, OrderBy
+
+from sentry import features, nodestore
+from sentry.api.endpoints.group_similar_issues_embeddings import get_stacktrace_string
+from sentry.eventstore.models import Event
+from sentry.grouping.grouping_info import get_grouping_info
+from sentry.issues.grouptype import ErrorGroupType
+from sentry.issues.occurrence_consumer import EventLookupError
+from sentry.models.group import Group
+from sentry.models.grouphash import GroupHash
+from sentry.models.project import Project
+from sentry.seer.utils import (
+    CreateGroupingRecordData,
+    CreateGroupingRecordsRequest,
+    post_bulk_grouping_records,
+)
+from sentry.silo.base import SiloMode
+from sentry.snuba.dataset import Dataset
+from sentry.snuba.referrer import Referrer
+from sentry.tasks.base import instrumented_task
+from sentry.utils import json, metrics, redis
+from sentry.utils.safe import get_path
+from sentry.utils.snuba import bulk_snuba_queries
+
+BATCH_SIZE = 20
+SEER_BACKFILL_DELAY_PER_RECORD = 0.1
+BACKFILL_NAME = "backfill_grouping_records"
+LAST_PROCESSED_REDIS_KEY = "grouping_record_backfill.last_processed_id"
+
+logger = logging.getLogger(__name__)
+
+
+class GroupEventRow(TypedDict):
+    event_id: str
+    group_id: int
+    message: str
+
+
+class GroupStacktraceData(TypedDict):
+    data: list[CreateGroupingRecordData]
+    stacktrace_list: list[str]
+
+
+@instrumented_task(
+    name="sentry.tasks.backfill_seer_grouping_records",
+    queue="default",
+    max_retries=0,
+    silo_mode=SiloMode.REGION,
+    soft_time_limit=60 * 15,
+    time_limit=60 * 15 + 5,
+)
+@metrics.wraps(f"{BACKFILL_NAME}.task")
+def backfill_seer_grouping_records(
+    project_id: int, last_processed_id: int, *args: Any, **kwargs: Any
+) -> None:
+    project = Project.objects.get_from_cache(id=project_id)
+    if not features.has("projects:similarity-embeddings-grouping", project):
+        return
+
+    redis_client = redis.redis_clusters.get(settings.SENTRY_MONITORS_REDIS_CLUSTER)
+    if last_processed_id is None:
+        last_processed_id = int(redis_client.get(LAST_PROCESSED_REDIS_KEY) or 0)
+
+    group_id_message_data_batch = (
+        Group.objects.filter(
+            project_id=project.id, id__gt=last_processed_id, type=ErrorGroupType.type_id
+        )
+        .values_list("id", "message", "data")
+        .order_by("id")[:BATCH_SIZE]
+    )
+    group_id_message_batch = {
+        group_id: message
+        for (group_id, message, data) in group_id_message_data_batch
+        if not get_path(data, "metadata", "embeddings_info", "nn_model_version")
+    }
+    batch_len = len(group_id_message_batch)
+    if batch_len == 0:
+        return
```

Comment:
This could cause this task to end early if all groups in the batch have `nn_model_version` set. Do we actually want to halt here? What if later groups don't have it set, or if we're wanting to re-run this?

This probably makes sense to do as part of `group_id_message_data_batch`?